File: app/src/test/java/com/myapplication/rick/packagedelivery/TestStreetClass.java
package com.myapplication.rick.packagedelivery;

import org.junit.Test;
import static org.junit.Assert.*;
/**
 * Created by Rick on 2-5-2016.
 */
public class TestStreetClass {
    private final String streetName;

    private final Range range;

    public TestStreetClass() {
        streetName  = "schoolstreet";
        final int lowerBoundStandard = 1;
        final int upperBoundStandard = 20;
        range = new Range(lowerBoundStandard, upperBoundStandard, RangeType.All);
    }
    @Test(expected = IllegalArgumentException.class)
    public void streetCreationWithNullRange() throws IllegalArgumentException {
        Range range = null;
        Street street = new Street(streetName, range);
    }


    @Test
    public void streetCreationValid() {
        Street street = new Street(streetName, range);
    }

    @Test
    public void checkStreetObjectParams() {
        Street street = new Street(streetName, range);
        assertEquals(streetName, street.getName());
        assertEquals(range, street.getRange());
    }

    @Test (expected = IllegalArgumentException.class)
    public void streetCreationNameWithOnlyWhitespaceShouldFail() throws IllegalArgumentException {
        String whitespace = "                              ";
        Street street = new Street(whitespace, range);
    }

    @Test (expected = IllegalArgumentException.class)
    public void streetCreationWithNullNameFails() throws IllegalArgumentException {
        Street street = new Street(null, range);
    }

    @Test
    public void toStringTest() {
        Street street = new Street(streetName, range);
        String expected = streetName + ", " + range.toString();
        assertEquals(expected, street.toString());
    }
}

File: app/src/test/java/com/myapplication/rick/packagedelivery/TestStreetClass.kt
package com.myapplication.rick.packagedelivery

import org.junit.Test
import org.junit.Assert.*

/**
 * Created by Rick on 2-5-2016.
 */
class TestStreetClass {
    private val streetName: String

    private val range: Range

    init {
        streetName = "schoolstreet"
        val lowerBoundStandard = 1
        val upperBoundStandard = 20
        range = Range(lowerBoundStandard, upperBoundStandard, RangeType.All)
    }

    @Test
    fun streetCreationValid() {
        val street = Street(streetName, range)
    }

    @Test
    fun checkStreetObjectParams() {
        val street = Street(streetName, range)
        assertEquals(streetName, street.name)
        assertEquals(range, street.range)
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(IllegalArgumentException::class)
    fun streetCreationNameWithOnlyWhitespaceShouldFail() {
        val whitespace = "                              "
        val street = Street(whitespace, range)
    }


    @Test
    fun toStringTest() {
        val street = Street(streetName, range)
        val expected = streetName + ", " + range.toString()
        assertEquals(expected, street.toString())
    }
}
